src/main/java/de/hs_mannheim/informatik/ct/util/DocxTemplate.java

```diff
@@ -80,11 +82,13 @@ public XWPFDocument generate(List<T> dataSource) throws IOException, XmlExceptio
         // Remove template page from final doc
         document.getDocument().setBody(new XWPFDocument().getDocument().getBody());
 
+
         // Generate new Pages
         val pageDataSource = StreamSupport.stream(
                 ZipPageData(dataSource, imageIds).spliterator(),
                 false).collect(Collectors.toList());
 
+//
         val templateXml = getParagraphTemplateXml(templateBuffer);
 
         pageDataSource
```

Comment:
Since you only have one room remaining in this method you can remove all of this, since it only makes sense when iterating multiple rooms.
``` java
   .parallelStream()
                .unordered()
                .map(data -> new Indexed<>(data.index, applyPageParagraphXml(templateXml, data.value)))
                .sorted(Comparator.comparingInt(o -> o.index))
                .forEachOrdered(paragraphs -> {
```